scipy/stats/_binned_statistic.py

```diff
@@ -203,38 +224,93 @@ def binned_statistic_2d(x, y, values, statistic='mean',
     bins : int or [int, int] or array_like or [array, array], optional
         The bin specification:
 
-          * the number of bins for the two dimensions (nx=ny=bins),
+          * the number of bins for the two dimensions (nx = ny = bins),
           * the number of bins in each dimension (nx, ny = bins),
           * the bin edges for the two dimensions (x_edges = y_edges = bins),
           * the bin edges in each dimension (x_edges, y_edges = bins).
 
+        If the bin edges are specified, the number of bins will be,
+        (nx = len(x_edges)-1, ny = len(y_edges)-1).
+
     range : (2,2) array_like, optional
         The leftmost and rightmost edges of the bins along each dimension
         (if not specified explicitly in the `bins` parameters):
         [[xmin, xmax], [ymin, ymax]]. All values outside of this range will be
         considered outliers and not tallied in the histogram.
+    expand_binnumbers : bool, optional
+        .. versionadded:: 0.17.0
+        'False' (default): the returned `binnumbers` is a shape (N,) array of
+        linearized bin indices.
+        'True': the returned `binnumbers` is 'unraveled' into a shape (2,N)
+        ndarray, where each row gives the bin numbers in the corresponding
+        dimension.
+        See the `binnumbers` returned value, and the `Examples` section.
 
     Returns
     -------
     statistic : (nx, ny) ndarray
-        The values of the selected statistic in each two-dimensional bin
+        The values of the selected statistic in each two-dimensional bin.
     x_edges : (nx + 1) ndarray
         The bin edges along the first dimension.
     y_edges : (ny + 1) ndarray
         The bin edges along the second dimension.
-    binnumber : 1-D ndarray of ints
-        This assigns to each observation an integer that represents the bin
-        in which this observation falls. Array has the same length as `values`.
+    binnumbers : (N,) array of ints or (D,N) ndarray of ints
+        This assigns to each element of `sample` an integer that represents the
+        bin in which this observation falls.  The representation depends on the
+        `expand_binnumbers` argument.
+        If 'False' (default): The returned `binnumbers` is a shape (N,) array
+        of linearized indices mapping each element of `sample` to its
+        corresponding bin (using row-major ordering).
+        If 'True': The returned `binnumbers` is a shape (2,N) ndarray where
+        each row indicates where the elements of `sample` should be inserted,
+        into the `x_edges` and `y_edges` arrays respectively, so as to keep
+        the arrays sorted.  In each dimension, a binnumber of `i` means the
+        corresponding value is between (D_edges[i-1], D_edges[i]), where 'D' is
+        either 'x' or 'y'.
+
```

Comment:
This strikes me as a bit long for a parameter description. I would put these details under ``Notes`` below.